packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte

```diff
@@ -62,15 +58,53 @@
 			return;
 		}
 
-		if (shouldUpdateUrl && !conversation.past_user_inputs.length) {
+		// Error checking
+		const config = model.config;
+		if (config === undefined) {
+			outputJson = "";
+			output = [];
+			error = "Model config not found";
+			return;
+		}
+
+		const tokenizerConfig = config.tokenizer as TokenizerConfig | undefined;
+		if (tokenizerConfig === undefined) {
+			outputJson = "";
+			output = [];
+			error = "Tokenizer config not found";
+			return;
+		}
+
+		const chatTemplate = tokenizerConfig.chat_template;
+		if (chatTemplate === undefined) {
+			outputJson = "";
+			output = [];
+			error = "No chat template found in tokenizer config";
+			return;
+		}
+
+		if (shouldUpdateUrl && !messages.length) {
 			updateUrl({ text: trimmedText });
 		}
 
+		// Add user message to chat
+		messages = [...messages, { role: "user", content: trimmedText }];
+
+		// Render chat template
+		const special_tokens_map = extractSpecialTokensMap(tokenizerConfig);
+
+		const template = new Template(chatTemplate);
+		const chatText = template.render({
+			messages,
+			add_generation_prompt: true,
+			...special_tokens_map,
+		});
+
 		const requestBody = {
```

Comment:
Could use the types from `@huggingface/tasks` :exploding_head:

Reply:
`requestBody` could be typed with `@huggingface/tasks`, for example here:
https://github.com/huggingface/huggingface.js/blob/ea2d4711685ddc42e966b725e29e62d7a4de2642/packages/tasks/src/tasks/text-generation/inference.ts#L10-L20

Let's keep it for a follow-up PR though

Reply:
Linking the related issues
#360 
#476